File: ClickDigitalFrontend/src/app/models/backend/log.ts
/**

 * This class represents a log in the view of the backend
 */
export class Log {
  date: Date;
  priority: string;
  source: string;
  username: string;
  action: string;
  object: string;
  status: string;

  constructor(obj: any) {
    if (obj.hasOwnProperty('date')) this.date = obj.date;
    if (obj.hasOwnProperty('priority')) this.priority = obj.priority;
    if (obj.hasOwnProperty('source')) this.source = obj.source;
    if (obj.hasOwnProperty('username')) this.username = obj.user;
    if (obj.hasOwnProperty('action')) this.action = obj.action;
    if (obj.hasOwnProperty('object')) this.object = obj.object;
    if (obj.hasOwnProperty('status')) this.status = obj.status;
  }
}

File: ClickDigitalBackend/src/main/java/acpmanager/logfilter/Log.java
package acpmanager.logfilter;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import org.apache.logging.log4j.Level;
import services.LogConstants;

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class Log {
    /**
     * date and time the log was made
     */
    String date = LogConstants.EMPTY_FIELD;
    /**
     * priority of the log
     */
    String priority = LogConstants.EMPTY_FIELD;
    /**
     * class where the log was made
     */
    String source = LogConstants.EMPTY_FIELD;
    /**
     * the associated username
     */
    String username = LogConstants.EMPTY_FIELD;
    /**
     * the action that was performed
     */
    String action = LogConstants.EMPTY_FIELD;
    /**
     * the affected object
     */
    String object = LogConstants.EMPTY_FIELD;
    /**
     * states if action was successful or if it failed
     */
    String status = LogConstants.EMPTY_FIELD;


    public Log(String date, String priority, String source, String username, String action, String object, String status) {

        this.date = date;
        this.priority = priority;
        this.source = source;
        this.username = username;
        this.action = action;
        this.object = object;
        this.status = status;
    }

    public Log(String username, String action, String object, String status) {
        this.username = username;
        this.action = action;
        this.object = object;
        this.status = status;
    }


    public Log() {

    }

    @Override
    public String toString() {
        return username + " " + action + " " + object + " " + status;
    }

}
